pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -342,13 +342,21 @@ def on_save_checkpoint(
             "best_model_path": self.best_model_path,
             "current_score": self.current_score,
             "dirpath": self.dirpath,
+            "best_k_models": self.best_k_models,
+            "kth_best_model_path": self.kth_best_model_path,
+            "kth_value": self.kth_value,
+            "last_model_path": self.last_model_path,
         }
 
     def on_load_checkpoint(
         self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", callback_state: Dict[str, Any]
     ) -> None:
         self.best_model_score = callback_state["best_model_score"]
         self.best_model_path = callback_state["best_model_path"]
+        self.best_k_models = callback_state["best_k_models"]
+        self.kth_best_model_path = callback_state["kth_best_model_path"]
+        self.kth_value = callback_state["kth_value"]
+        self.last_model_path = callback_state["last_model_path"]
```

Comment:
This is required for backwards compatibility with legacy checkpoints who don't have this.

```suggestion
        self.best_k_models = callback_state.get("best_k_models", self.best_k_models)
        self.kth_best_model_path = callback_state.get("kth_best_model_path", self.kth_best_model_path)
        self.kth_value = callback_state.get("kth_value", self.kth_value)
        self.last_model_path = callback_state.get("last_model_path", self.last_model_path)
```

Reply:
Thanks !